rfcs/20221025-sync-devices.md

```diff
@@ -0,0 +1,91 @@
+# API to synchronize devices
+
+| Status        | Proposed                                             |
+:-------------- |:---------------------------------------------------- |
+| **RFC #**     | [434](https://github.com/tensorflow/community/pull/434) |
+| **Author(s)** | Reed Wanderman-Milne (reedwm@google.com), Jonathan Dekhtiar (jdekhtiar@nvidia.com) |
+| **Sponsor**   | Rohan Jain (rohanj@google.com)                       |
+| **Updated**   | 2022-10-25                                           |
+
+## Objective
+
+This document proposes a simple API to synchronize TensorFlow devices: `tf.sync_devices()`. This is important in accurately measuring execution time in TensorFlow GPU benchmarks, especially in microbenchmarks.
```

Comment:
Can we put into this namespace. tf.config.experimental.sync_devices()?

Reply:
@sun51 It is not a config call. It could be call at every iterations:

```python
for step in range(NUM_STEPS):
   data = ds_iter.next()
   start_t = time.time()
   rslt = model(data)
   tf.sync_devices()
   print(f"Time: {time.time() - start_t}")
```

I don't think `config` is the appropriate namespace for such an API.

Reply:
It is actually changing the setting of runtime.  Meanwhile,  I am still not quite clear about the semantics of this API.  In the above case, model() will include many ops,  it is called  before the tf.sync_device() is even called. How does this work?   Once you sync device, can you set to normal config(async) later?

Reply:
I think you there is a misunderstanding on how the API works.
> It is actually changing the configuration of runtime.

No, the API introduces a blocking call that awaits for the OPs scheduled on every devices to be cleared. There is no "configuration" being updated or modified. It's just an API call that awaits for the GPUs to be done. Hence this has nothing to do in the `config` namespace IMHO.

Reply:
chatted with reed, put into tf.test.XXX namespace is something we both think reasonable.

Reply:
Sorry for the delay, I edited the RFC to put under tf.test.sync_devices.